backend/tessend/tessend.go

```diff
@@ -133,42 +143,80 @@ func (t *Tessend) handleWatchEvent(watchEvent store.WatchEventTessenConfig) {
 	t.interrupt <- tessen
 }
 
-// start starts a new scheduler for tessen
-func (t *Tessend) start(tessen *corev2.TessenConfig) error {
-	// Guard against updates while the daemon is shutting down
-	if err := t.ctx.Err(); err != nil {
-		return err
+// startRingUpdates starts a loop to periodically update the ring.
+func (t *Tessend) startRingUpdates() {
+	ticker := time.NewTicker(450 * time.Second)
```

Comment:
Sort of haha. We have 2 hardcoded intervals here, which really just determines how long we consider a ring entry alive (900 seconds) and how often we should update the ring for added/removed cluster members (450 seconds). Both of these checks need to occur more frequently than the tessen interval (1800 seconds). If we used `t.interval/2` and `t.interval/4`, there's a chance that if the tessen interval was ever updated in the response header, it could yield an unexpected result. Do you think it's safer to calculate these intervals rather than hardcode them?

Reply:
I think it's fine that they are hard coded, but let's pull them out into constants so they are clearly documented.

Reply:
The low bound (and the default value) for the Tessen update interval is 60 seconds though; do these liveness and update intervals (900s and 450s respectively) make sense? I honestly have no idea and I'll ultimately defer to you guys, but @nikkiki just said:

> Both of these checks need to occur more frequently than the tessen interval

Reply:
The upper bound and lower bound values are set to protect us from ourselves. Their only purpose is to validate that the Tessen API doesn't return a bogus response header (for example if we accidentally set the response header to 2,000,000 TessenD would not accept that value because its not within the upper/lower bound range).

The liveness and update intervals operate independently of that, and are merely values we chose for the ring to operate/update in the background.

Let's hop on a call if that's still not clear so I can better explain 😄